### ISSUE TITLE
Question about the use of `logits` for protein and `last_hidden_state` for drug representations

### ISSUE THREAD

**Issue by yyly6** [ZhaohanM/FusionDTI#5]
Hi! First of all, thank you for sharing your amazing work. I have been using your code and came across a part that I am a bit confused about. Specifically, in the `encoding` function, I noticed that the protein output is given as `logits` (shape: [64, 512, 446]), while the drug output is given as `last_hidden_state` (shape: [64, 512, 768]). Here's the relevant code:

```python
def encoding(self, prot_input_ids, prot_attention_mask, drug_input_ids, drug_attention_mask): 
    # Process inputs through encoders
    prot_embed = self.prot_encoder(
        input_ids=prot_input_ids, attention_mask=prot_attention_mask, return_dict=True
    ).logits
    # prot_embed = self.prot_reg(prot_embed)

    drug_embed = self.drug_encoder(
        input_ids=drug_input_ids, attention_mask=drug_attention_mask, return_dict=True
    ).last_hidden_state

    return prot_embed, drug_embed
```
Protein output: prot_embed = self.prot_encoder(...).logits (shape: [64, 512, 446])
Drug output: drug_embed = self.drug_encoder(...).last_hidden_state (shape: [64, 512, 768])

Could you explain the reasoning behind this design choice?

**Comment by ZhaohanM** [comment]
I appreciate your interest in our work. The purpose of our encoding function is to obtain token-level representations (`last_hidden_state` ) from drug and protein encoders. Sorry for the confusion, the final version of the code is shown below:

```python
    def encoding(self, prot_input_ids, prot_attention_mask, drug_input_ids, drug_attention_mask):
        # Process protein encoder with hidden state output
        prot_embed = self.prot_encoder(
            input_ids=prot_input_ids, 
            attention_mask=prot_attention_mask, 
            output_hidden_states=True,  # Request hidden states
            return_dict=True
        ).hidden_states[-1]

        drug_embed = self.drug_encoder(
            input_ids=drug_input_ids, attention_mask=drug_attention_mask, return_dict=True
        ).last_hidden_state

        return prot_embed, drug_embed
```

 Here's the detailed explanation:

1. **Drug Encoder (`drug_encoder`)**:
   - The `last_hidden_state` directly provides contextualised token-level representations.
   - The shape is typically `[batch_size, seq_length, hidden_size]`([64, 512, 768]).

2. **Protein Encoder (`prot_encoder`)**:
   - By setting `output_hidden_states=True` and extracting `hidden_states[-1]`, we obtain the last hidden state, which serves as the contextualised token-level representation.
   - The shape is also `[batch_size, seq_length, hidden_size]`([64, 512, 1280]).

**Comment by yyly6** [comment]
Thank you for your correction. You also should correct prot_embedding_shape:
```python
model = FusionDTI(446, 768, args).to(device)
```
change 446 to 1280

**Comment by ZhaohanM** [comment]
> Thank you for your correction. You also should correct prot_embedding_shape:
> 
> ```python
> model = FusionDTI(446, 768, args).to(device)
> ```
> 
> change 446 to 1280

Thanks for your reminder. The code has been updated.
